adapter/network_management.md

```diff
@@ -0,0 +1,253 @@
+* At any time, upon receiving `cmd.network.get_all_nodes` command, an adapter must respond with `evt.network.all_nodes_report` report containing status of all things.
+* On its own, an adapter must publish `evt.network.node_report` whenever the status of a thing changes.
+* On its own, an adapter should publish `evt.network.all_nodes_report` at least once per day.
+
+### Resetting Network
+
+* Upon receiving `cmd.network.reset` command, the adapter should start resetting the network.
+* For each removed thing it should emit `evt.thing.exclusion_report` as specified in [thing management](/adapter/thing_management.md#interfaces) document.
+* Once the process is finished, the adapter should emit `evt.network.reset_done`.
+
+## FIMP Specification
+
+### Topics
+
+Every adapter should have unique resource name and use the following topics in accordance with [topic format](/fimp/topic_format.md):
+
+`pt:j1/mt:cmd/rt:ad/rn:{RESOURCE_NAME}/ad:1`
+
+`pt:j1/mt:evt/rt:ad/rn:{RESOURCE_NAME}/ad:1`
+
+### Service name
+
+Every adapter should define its own service name.
+
+### Interfaces
+
+| Type | Interface                    | Value type | Description                                                                                                       |
+|------|------------------------------|------------|-------------------------------------------------------------------------------------------------------------------|
+| in   | cmd.network.get_node         | string     | Requests state of a thing with the provided address.                                                              |                         
+| out  | evt.network.node_report      | object     | Reports state of a single thing, see [`node_report`](#definitions) object.                                        |
+| in   | cmd.network.get_all_nodes    | null       | Requests state of all things.                                                                                     |                         
+| out  | evt.network.all_nodes_report | object     | Reports state of all things, an array of [`node_report`](#definitions) objects.                                   |
+| in   | cmd.network.update           | string     | Requests update of the network topology, see [`network_update_mode`](#definitions) definition for allowed values. | 
+| in   | cmd.network.node_update      | int        | Requests an interview and discovery of its capabilities for the provided address of a thing.                      |
+| in   | cmd.network.reset            | null       | Requests forceful removal of all things and resets the network.                                                   |
+| out  | evt.network.reset_done       | null       | Reports that network reset has been completed.                                                                    |
+
+### Definitions
+
+* `node_report` is an object with the following structure:
```

Comment:
Issue 2: `node_report` object copies verbatim 5 values  from the inclusion report, which is redundant. However, 3 of these values, even though they are stored verbatim in Vinculum DB, are not processed and not provided by Vinculum in thing/device registry reports. As such they are not forwarded by Cloud Bridge to Niflheim/Site Structure. In essence they are unavailable for mobile application nor any application running on the hub.

We could of course wait for Vinculum,CloudBridge and Niflheim to be updated, but due to update constraints it seems unrealistic and impossible to meet the general timetable for signal quality monitoring features.

Comment if you have better ideas.

Reply:
Zwave ad can extend this report with needed values.

Reply:
EDIT: But some properties are already available for SDU App: hash, alias?
Could You point out which properties are propagated to site structure and which are not?

Reply:
Yes, but ThingsplexUI utilizes hash and alias from this report. Nothing else is available in site structure.

Reply:
I meant, that I can clearly see e.g. alias for the whole lifetime of a device in SDU App->device details section... Isn't that enough? I mean, if it's visible there then app has to have it stored somewhere?